mailbox/store/src/main/java/org/apache/james/mailbox/store/MessageStorer.java

```diff
@@ -46,10 +46,18 @@
 
 public interface MessageStorer {
     /**
-     * If applicable, this method will parse the messageContent to retrieve associated attachments and will store them.
+     * If supported by the underlying implementation, this method will parse the messageContent to retrieve associated
+     * attachments and will store them.
+     *
+     * Otherwize an empty optional will be returned on the right side of the pair.
      */
     Pair<MessageMetaData, Optional<List<MessageAttachmentMetadata>>> appendMessageToStore(Mailbox mailbox, Date internalDate, int size, int bodyStartOctet, SharedInputStream content, Flags flags, PropertyBuilder propertyBuilder, MailboxSession session) throws MailboxException;
```

Comment:
if it's at the implementation level, why not `Optional<Pair<MessageMetaData, List<MessageAttachmentMetadata>>` ?

Reply:
You always have the message metadata but the underlying implementation might not return you attachment metadata (because it does not support it)